src/parser/parser.c

```diff
@@ -1,37 +1,46 @@
 #include "cub3D.h"
 
+struct s_parser_state
```

Comment:
didnt we want to use typedefs all the time?

Reply:
this state struct is yet to be extended for all the parsing states etc. Up until it is just in use in this file I didn't see a point in making it a typedef in the header file